addons/sale/models/sale_order.py

```diff
@@ -307,6 +308,16 @@ def _compute_display_name(self):
                 name = f'{name} - {order.partner_id.name}'
             order.display_name = name
 
+    @api.depends('state', 'order_line.product_id')
+    def _compute_has_archived_product_line(self):
+        for order in self:
+            if order.state in ['draft', 'sent']:
```

Comment:
Honestly, since the field name is `has_archived_product_line` (and I can imagine some flows where one customer would like to highlight archived products even on confirmed orders), I believe the field should not depend on the sale order state (but in the view, by default, you would hide the warning if the SO/SOL is not in draft/sent state).

Reply:
ohh actually I make field [display_archived_product_warning](https://github.com/odoo/odoo/pull/157208#discussion_r1555730487) but it was looking like a settings field name

Got it! Done :100: